### PR TITLE
[Compile Time Constant Extraction] Ensure macro-expanded declarations are visited in WMO

### PR DESCRIPTION
**Pull request by artemcm**
Resolves rdar://130036855

### CODE REVIEW

**Comment by artemcm** [comment]
@swift-ci test

File: test/ConstExtraction/ExtractFromMacroExpansion.swift [hunk 1]
```diff
@@ -7,6 +7,10 @@
 // RUN: %target-swift-frontend -typecheck -emit-const-values-path %t/ExtractFromMacroExpansion.swiftconstvalues -const-gather-protocols-file %t/protocols.json -primary-file %s -load-plugin-library %t/%target-library-name(MacroDefinition)
 // RUN: cat %t/ExtractFromMacroExpansion.swiftconstvalues 2>&1 | %FileCheck %s
 
+// Do the same, but ensure the WMO compilation flow produces the same result
+// RUN: %target-swift-frontend -typecheck -emit-const-values-path %t/ExtractFromMacroExpansionWMO.swiftconstvalues -const-gather-protocols-file %t/protocols.json -O %s -load-plugin-library %t/%target-library-name(MacroDefinition)
```
Comment (ApolloZhu) [inline test/ConstExtraction/ExtractFromMacroExpansion.swift:11]: Do we know for sure that -O will always enable WMO?

Comment (artemcm) [reply test/ConstExtraction/ExtractFromMacroExpansion.swift:11]: It's not actually `-O` here, rather it is the absence of `-primary-file` directives, which triggers WMO and the new code-path.

**Review by ApolloZhu — APPROVED** [review]
Implementation looks reasonable

**Pull request merged**